### ISSUE TITLE
bug: GLM-OCR returns empty content on CUDA (layout-paired path bypasses inference)

### ISSUE THREAD

**Issue by Goldziher** [kreuzberg-dev/kreuzberg#1175]
## Summary
GLM-OCR returns **empty content** (Ok, no error/panic/assert) on NVIDIA GPU (`candle-cuda`) while producing correct "Hello World" on CPU. Reproduced on the L4 (x86_64) runner in CI GPU run [28326108768](https://github.com/xberg-io/xberg/actions/runs/28326108768), test `candle_glm_ocr_e2e_extraction` (`crates/xberg/tests/candle_backends.rs:503`).

## Key observation
With `XBERG_GLM_DEBUG=1` + `--nocapture`, **zero** `[glm-debug]` lines were emitted — meaning the instrumented `process_image_inner` (in `glm_ocr/mod.rs`) was **never reached**.

## Root cause (primary hypothesis)
`GlmOcrBackend` is constructed with `LayoutMode::default()` = **Paired**, which routes through `process_paired()`:

```
GlmOcrBackend::process_image                         (glm_ocr_backend.rs:418)
  → process_paired()                                 (glm_ocr_backend.rs:557)
    → PP-DocLayout-V3 layout inference  → regions
    → for region in sorted_regions { ... }           (glm_ocr_backend.rs:573)
        → engine.process_image_with_task()           (line 595)  ← only if regions > 0
    → parts.join("\n\n")                              (line 647)
```

When **layout detection returns zero regions**, the for-loop never iterates, no GLM inference runs, and `parts.join()` yields an empty string. This explains both the empty content **and** the absent debug output. Layout detection (PP-DocLayout-V3) runs via ONNX Runtime, whose CUDA EP can behave differently than CPU — a plausible source of 0-region output on GPU for this fixture.

## Secondary hypothesis (if regions are detected)
CUDA divergence in the generation sampling loop — `crates/xberg-candle-ocr/src/models/glm_ocr/mtp.rs`:
- softmax hard-coded over dim 0 (lines ~271-272) — wrong if CUDA logits shape differs from CPU `(1, vocab)`
- `argmax(0)` greedy path (lines ~245-250)
Either could make the first sampled token EOS ([59246, 59253]) → zero content tokens.

## Action
1. Add layout-region-count + prefill-logits-shape diagnostics (eprintln, `XBERG_GLM_DEBUG`-gated) at the bypass point and in `mtp.rs`, re-dispatch CI GPU to confirm which hypothesis holds.
2. Fix accordingly (layout CUDA EP / region fallback, or sampling-shape correction).

## Scope
- `crates/xberg/src/candle_ocr/glm_ocr_backend.rs` (process_paired / region loop)
- `crates/xberg-candle-ocr/src/models/glm_ocr/mtp.rs` (sampling)
- Validate only on the GPU runner (not reproducible without CUDA).

## Acceptance
`Candle GPU (glm-ocr)` job green in CI GPU.
